transport/grpc/config.go

```diff
@@ -78,9 +80,43 @@ type TransportConfig struct {
 // inbounds:
 //   grpc:
 //     address: ":80"
+//     tls:
+//       enabled: true
+//       keyFile: "/path/to/key"
+//       certFile: "/path/to/cert"
 type InboundConfig struct {
 	// Address to listen on. This field is required.
-	Address string `config:"address,interpolate"`
+	Address string           `config:"address,interpolate"`
+	TLS     InboundTLSConfig `config:"tls,optional"`
+}
+
+func (c InboundConfig) inboundOptions() ([]InboundOption, error) {
+	return c.TLS.inboundOptions()
+}
+
+// InboundTLSConfig configures a gRPC inbound TLS credentials.
+type InboundTLSConfig struct {
+	Enabled  bool   `config:"enabled,optional"`
+	CertFile string `config:"certFile,optional"`
+	KeyFile  string `config:"keyFile,optional"`
+}
+
+func (c InboundTLSConfig) inboundOptions() ([]InboundOption, error) {
+	if c.Enabled {
```

Comment:
nit: you can invert the nesting here for slightly better readability:

```
if !c.Enabled {
  return nil,nil
}
// The rest of it
```

Reply:
done.

Reply:
good catch - fixed elsewhere